### ISSUE TITLE
require_function_arguments() silently ignores nested calls (descendant-axis false negative) [na-1]

### ISSUE THREAD

**Issue by fabiandistler** [fabiandistler/lintrhelper#57]
> *This was generated by AI during triage.*

Split out from #52 (audit finding `na-1`) during triage. The audit reports an executed repro; code paths below were statically confirmed on `main`, repros were not re-executed here.

## Current behavior

`require_function_arguments()` collects named arguments with the descendant axis `.//SYMBOL_SUB`, so an argument belonging to a *nested* call satisfies the requirement of the outer call. No error, no warning, no lint — a silent false negative:

```r
explicit_saf <- require_function_arguments("data.frame", "stringsAsFactors")
# on `data.frame(a = read.csv(p, stringsAsFactors = FALSE))` -> 0 lints, expected 1
```

This hits exactly the documented example pattern of the function (roxygen example with `data.frame` / `stringsAsFactors`). At the same node, `.//SYMBOL_SUB` yields `a, stringsAsFactors` while `./SYMBOL_SUB` yields only `a`.

## Desired behavior

Only the call's own (direct-child) named arguments count toward the requirement. The nested-call case above produces 1 lint; flat calls behave exactly as today.

## Fix sketch (from audit)

Use the child axis `./SYMBOL_SUB` instead of `.//SYMBOL_SUB` where the call's own argument list is meant. Positional passing and partial matching (`stringsAsFact = FALSE`) produce no (or a differently-texted) `SYMBOL_SUB` and are treated equally by both axes, so the fix changes only the nested case. Add both nested cases as regression tests — the existing suite (`tests/testthat/test-high_level_linters.R`, flat-call block) covers only flat calls.

## Severity

High per finder, corrected to medium by the audit's impact review: a style rule that silently doesn't fire on ordinary code, in an exported README-advertised API with no test coverage — more than cosmetic, no data/persistence consequences.

## Related

- #52 (source audit, to be closed after extraction)
- #32 (high-level helpers rebuild — overlaps thematically)

**Comment by fabiandistler** [comment]
> *This was generated by AI during triage.*

## Verification

Reproduced from the reporter steps on `main` (`devtools::load_all`):

- `df <- data.frame(x = 1:3)` -> 1 lint (correct)
- `df <- data.frame(x = 1:3, stringsAsFactors = FALSE)` -> 0 lints (correct)
- `data.frame(a = read.csv(p, stringsAsFactors = FALSE))` -> **0 lints, expected 1 — bug confirmed**

The named-argument lookup uses the descendant axis, so a `SYMBOL_SUB` belonging to a nested call satisfies the outer call requirement. No grilling needed — the issue is fully specified.

## Agent Brief

**Category:** bug
**Summary:** `require_function_arguments()` lets a nested call satisfy the outer call argument requirement

**Current behavior:**
A linter built by `require_function_arguments(function_name, required_args)` collects named arguments with a descendant search, so an argument belonging to a *nested* call counts toward the outer call. `data.frame(a = read.csv(p, stringsAsFactors = FALSE))` produces 0 lints even though the outer `data.frame()` call never passes `stringsAsFactors` — a silent false negative on the documented example pattern.

**Desired behavior:**
Only the call's own (direct) named arguments count toward the requirement. The nested-call case above produces 1 lint on the outer call; flat calls behave exactly as today. Positional passing and partial-name matches keep their current semantics.

**Key interfaces:**
- `require_function_arguments(function_name, required_args, message, type)` — the builder whose returned linter performs the named-argument matching; the matching logic needs to consider only the call's own argument list, not arguments of calls nested inside it
- The existing flat-call test block for this builder — extend with nested-call regression cases rather than replacing it

**Acceptance criteria:**
- [ ] `data.frame(x = 1:3)` still produces 1 lint
- [ ] `data.frame(x = 1:3, stringsAsFactors = FALSE)` still produces 0 lints
- [ ] `data.frame(a = read.csv(p, stringsAsFactors = FALSE))` produces 1 lint (the reported bug)
- [ ] A nested call where the *outer* call does supply the argument still passes (no new false positive)
- [ ] The existing test suite for the high-level linters passes

**Out of scope:**
- Positional-argument or partial-name matching semantics
- Other high-level linter builders or any MCP/plumbing changes
- Documentation beyond what the fix itself requires
